src/Test/TestAIMuriel.java

```diff
@@ -55,37 +60,65 @@ private void cleanupTest()
   public boolean runTest()
   {
     boolean testPassed = true;
-    testPassed &= validate(testBuildMegatank(), "  Build Megatank test failed.");
-    testPassed &= validate(testHuntStall(), "  Hunting stall-test failed.");
-    testPassed &= validate(testClearAttackRoute(), "  Route clearing test failed.");
-    testPassed &= validate(testWalkInLine(), "  Line walking test failed.");
-    testPassed &= validate(testInfWadeThroughTanks(), "  Infantry move priority test failed.");
-    testPassed &= validate(testTankWadeThroughInfs(), "  Tank move priority test failed.");
+    testPassed &= validate(testForAllAIs((ai) -> testBuildMegatank(ai)), "  Build Megatank test failed.");
```

Comment:
Really?
Look, I get that lambdas are like, the most excellent thing ever invented in the history of programming, but here is yet another case where it probably isn't warranted.

Granted, it's super cool to be able to just say "run this test on all the AIs", but on the other hand, that ties their success together. Which, if I recall the initial vision for this capability, is exactly not what we want. We want to set per-AI thresholds (Muriel must pass 100% of existing tests, Wally must pass 85%, ISAI must pass 2%, whatever) so that we can catch regressions when we go tweaking things.

And I know you are now going to go and simply reverse your lambda, but really - a simple loop is all we need here.

Reply:
I suppose it would be roughly as concise (and probably more clear, really) to just wrap the tests in a loop.